OpenCL compile failure with `Werror-1` on Raspberry Pi 5 / Pi OS

```
g++ -o build/src/runtime/CL/functions/CLBitwiseOr.o -c -DARCH_ARM -Wextra -Wdisabled-optimization -Wformat=2 -Winit-self -Wstrict-overflow=2 -Wswitch-default -Woverloaded-virtual -Wformat-security -Wctor-dtor-privacy -Wsign-promo -Weffc++ -Wno-overlength-strings -Wall -std=c++14 -pedantic -Wlogical-op -Wnoexcept -Wstrict-null-sentinel -Wno-misleading-indentation -C -Wno-ignored-attributes -DENABLE_FP32_KERNELS -DENABLE_QASYMM8_KERNELS -DENABLE_QASYMM8_SIGNED_KERNELS -DENABLE_QSYMM16_KERNELS -DENABLE_INTEGER_KERNELS -DENABLE_NHWC_KERNELS -DENABLE_NCHW_KERNELS -Werror -march=armv8-a -O3 -DARM_COMPUTE_CPP_SCHEDULER=1 -DENABLE_NEON -DARM_COMPUTE_ENABLE_NEON -DARM_COMPUTE_CPU_ENABLED -DARM_COMPUTE_OPENCL_ENABLED -DEMBEDDED_KERNELS -DARM_COMPUTE_VERSION_MAJOR=53 -DARM_COMPUTE_VERSION_MINOR=2 -DARM_COMPUTE_VERSION_PATCH=0 -Iinclude -I. -Ibuild/build/src/core/CL -Ibuild/src/core/CL -Ibuild/build/src/core -Ibuild/src/core -Ibuild/src/cpu/kernels/assembly -Isrc/cpu/kernels/assembly -Ibuild/src/core/NEON/kernels/arm_gemm -Isrc/core/NEON/kernels/arm_gemm -Ibuild/src/core/NEON/kernels/convolution/common -Isrc/core/NEON/kernels/convolution/common -Ibuild/src/core/NEON/kernels/convolution/winograd -Isrc/core/NEON/kernels/convolution/winograd -Ibuild/src/core/NEON/kernels/arm_conv/depthwise -Isrc/core/NEON/kernels/arm_conv/depthwise -Ibuild/src/core/NEON/kernels/arm_conv/pooling -Isrc/core/NEON/kernels/arm_conv/pooling -Ibuild/src/core/NEON/kernels/arm_conv -Isrc/core/NEON/kernels/arm_conv -Ibuild/src/core/NEON/kernels/assembly -Isrc/core/NEON/kernels/assembly -Ibuild/arm_compute/core/NEON/kernels/assembly -Iarm_compute/core/NEON/kernels/assembly -Ibuild/src/cpu/kernels/assembly -Isrc/cpu/kernels/assembly -Ibuild/third_party/kleidiai -Ithird_party/kleidiai src/runtime/CL/functions/CLBitwiseOr.cpp
In file included from /usr/include/c++/14/deque:66,
                 from src/runtime/CL/mlgo/HeuristicTree.cpp:31:
In member function ‘void std::_Deque_base<_Tp, _Alloc>::_M_create_nodes(_Map_pointer, _Map_pointer) [with _Tp = long unsigned int; _Alloc = std::allocator<long unsigned int>]’,
    inlined from ‘void std::_Deque_base<_Tp, _Alloc>::_M_initialize_map(std::size_t) [with _Tp = long unsigned int; _Alloc = std::allocator<long unsigned int>]’ at /usr/include/c++/14/bits/stl_deque.h:658:19,
    inlined from ‘void std::deque<_Tp, _Alloc>::_M_range_initialize(_ForwardIterator, _ForwardIterator, std::forward_iterator_tag) [with _ForwardIterator = const long unsigned int*; _Tp = long unsigned int; _Alloc = std::allocator<long unsigned int>]’ at /usr/include/c++/14/bits/deque.tcc:448:25,
    inlined from ‘std::deque<_Tp, _Alloc>::deque(std::initializer_list<_Tp>, const allocator_type&) [with _Tp = long unsigned int; _Alloc = std::allocator<long unsigned int>]’ at /usr/include/c++/14/bits/stl_deque.h:980:21,
    inlined from ‘bool arm_compute::mlgo::HeuristicTree::check_if_structurally_correct() const’ at src/runtime/CL/mlgo/HeuristicTree.cpp:179:38:
/usr/include/c++/14/bits/stl_deque.h:683:40: error: assuming pointer wraparound does not occur when comparing P +- C1 with P +- C2 [-Werror=strict-overflow]
  683 |           for (__cur = __nstart; __cur < __nfinish; ++__cur)
      |                                  ~~~~~~^~~~~~~~~~~
g++ -o build/src/gpu/cl/kernels/ClHeightConcatenateKernel.o -c -DARCH_ARM -Wextra -Wdisabled-optimization -Wformat=2 -Winit-self -Wstrict-overflow=2 -Wswitch-default -Woverloaded-virtual -Wformat-security -Wctor-dtor-privacy -Wsign-promo -Weffc++ -Wno-overlength-strings -Wall -std=c++14 -pedantic -Wlogical-op -Wnoexcept -Wstrict-null-sentinel -Wno-misleading-indentation -C -Wno-ignored-attributes -DENABLE_FP32_KERNELS -DENABLE_QASYMM8_KERNELS -DENABLE_QASYMM8_SIGNED_KERNELS -DENABLE_QSYMM16_KERNELS -DENABLE_INTEGER_KERNELS -DENABLE_NHWC_KERNELS -DENABLE_NCHW_KERNELS -Werror -march=armv8-a -O3 -DARM_COMPUTE_CPP_SCHEDULER=1 -DENABLE_NEON -DARM_COMPUTE_ENABLE_NEON -DARM_COMPUTE_CPU_ENABLED -DARM_COMPUTE_OPENCL_ENABLED -DEMBEDDED_KERNELS -DARM_COMPUTE_VERSION_MAJOR=53 -DARM_COMPUTE_VERSION_MINOR=2 -DARM_COMPUTE_VERSION_PATCH=0 -Iinclude -I. -Ibuild/build/src/core/CL -Ibuild/src/core/CL -Ibuild/build/src/core -Ibuild/src/core -Ibuild/src/cpu/kernels/assembly -Isrc/cpu/kernels/assembly -Ibuild/src/core/NEON/kernels/arm_gemm -Isrc/core/NEON/kernels/arm_gemm -Ibuild/src/core/NEON/kernels/convolution/common -Isrc/core/NEON/kernels/convolution/common -Ibuild/src/core/NEON/kernels/convolution/winograd -Isrc/core/NEON/kernels/convolution/winograd -Ibuild/src/core/NEON/kernels/arm_conv/depthwise -Isrc/core/NEON/kernels/arm_conv/depthwise -Ibuild/src/core/NEON/kernels/arm_conv/pooling -Isrc/core/NEON/kernels/arm_conv/pooling -Ibuild/src/core/NEON/kernels/arm_conv -Isrc/core/NEON/kernels/arm_conv -Ibuild/src/core/NEON/kernels/assembly -Isrc/core/NEON/kernels/assembly -Ibuild/arm_compute/core/NEON/kernels/assembly -Iarm_compute/core/NEON/kernels/assembly -Ibuild/src/cpu/kernels/assembly -Isrc/cpu/kernels/assembly -Ibuild/third_party/kleidiai -Ithird_party/kleidiai src/gpu/cl/kernels/ClHeightConcatenateKernel.cpp
cc1plus: all warnings being treated as errors
scons: *** [build/src/runtime/CL/mlgo/HeuristicTree.o] Error 1
scons: building terminated because of errors.
```

How to reproduce:

```
 git clone https://github.com/ARM-software/ComputeLibrary.git
 cd ComputeLibrary/
 git checkout v53.2.0 
 scons Werror=1 -j8 debug=0 neon=1 opencl=1 os=linux arch=armv8a build=native 2>&1 | tee build.log
```

Attaching zipfile of full build log.

[build.zip](https://github.com/user-attachments/files/30061304/build.zip)

--- a/src/runtime/CL/mlgo/HeuristicTree.cpp
+++ b/src/runtime/CL/mlgo/HeuristicTree.cpp
@@ -28,8 +28,8 @@
 #include "support/Cast.h"
 
 #include <algorithm>
-#include <deque>
 #include <set>
+#include <vector>
 namespace arm_compute
 {
 namespace mlgo
@@ -175,13 +175,15 @@ bool HeuristicTree::add_branch(NodeID id, Condition cond, NodeID t_node, NodeID 
 
 bool HeuristicTree::check_if_structurally_correct() const
 {
-    std::set<NodeID>   visited;
-    std::deque<NodeID> to_visit{_root};
+    std::set<NodeID>    visited;
+    std::vector<NodeID> to_visit;
+    to_visit.push_back(_root);
+    std::size_t front = 0;
 
-    while (!to_visit.empty())
+    while (front < to_visit.size())
     {
-        auto id = to_visit.front();
-        to_visit.pop_front();
+        auto id = to_visit[front];
+        ++front;
         if (_tree.find(id) == _tree.end())
         {
             ARM_COMPUTE_LOG_INFO_MSG_WITH_FORMAT_CORE("Missing node %zu", id);

